x-pack/plugins/maps/index.js

```diff
@@ -11,11 +11,12 @@ import mappings from './mappings.json';
 import { checkLicense } from './check_license';
 import { watchStatusAndLicenseToInitialize } from
   '../../server/lib/watch_status_and_license_to_initialize';
+import { initTelemetryCollection } from './server/maps_telemetry';
 
 export function maps(kibana) {
 
   return new kibana.Plugin({
-    require: ['kibana', 'elasticsearch', 'xpack_main', 'tile_map'],
+    require: ['kibana', 'elasticsearch', 'xpack_main', 'tile_map', 'task_manager'],
```

Comment:
Hi, just to let you know, if someone needed to disable task manager, they would also have to disable the Maps plugin.

You could avoid that by putting the task stuff in `x-pack/plugins/oss_telemetry`

Reply:
After some offline conversation and some code digging, I've decided to leave this as is for now with the understanding that we may need to rethink this dependency in the future. There shouldn't be a timing issue as the plugin spec requires dependencies be both enabled and initialized (https://github.com/elastic/kibana/blob/94d99d63f3157d4a08eea83aa2489158b66a5e87/src/plugin_discovery/plugin_spec/plugin_spec.js#L35) however there is the issue of disabling the task manager effectively disabling the map. It may be possible to do a conditional require, however the condition would usually be on `xpack.task_manager.enabled` which is obtained from the server config at init time, trying to get this info at the point of require feels hacky. This might be a bigger issue of how to elegantly handle conditional requires in a broader way moving forward and could likely be tackled in the plugin spec.